### PR TITLE
fix(dop,gittar): fix some bug of dev work flow

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
gittar: Branch merge send event
dop: Only the default pipeline is supported

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMjgwXSwiYXNzaWduZWUiOlsiMTAwMTI2MSJdfQ%3D%3D&id=319010&iterationID=1280&pId=0&type=BUG)
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMjgwXSwiYXNzaWduZWUiOlsiMTAwMTI2MSJdfQ%3D%3D&id=319063&iterationID=1280&pId=0&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     fix some bug of dev work flow         |
| 🇨🇳 中文    |      解决研发工作流的一些缺陷        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/apps/devflow/flow/flow.service.go [hunk 1]
```diff
@@ -917,79 +922,26 @@ func (s *Service) getAllPipelineYml(ctx context.Context, app *apistructs.Applica
...
+	pipelineTreeData, err := s.fileTree.ListFileTreeNodes(apistructs.UnifiedFileTreeNodeListRequest{
+		Scope:   apistructs.FileTreeScopeProjectApp,
+		ScopeID: strconv.FormatUint(app.ProjectID, 10),
+		Pinode:  base64.URLEncoding.EncodeToString([]byte(defaultYmlPath)),
+		IdentityInfo: apistructs.IdentityInfo{
+			UserID: apis.GetUserID(ctx),
+		},
+	}, uint64(orgID))
 	if err != nil {
+		s.p.Log.Debug("failed to get %v path yml error %v", apistructs.DefaultPipelineYmlName, err)
```
Comment (sfwn) [inline internal/apps/devflow/flow/flow.service.go:935]: use error level

File: internal/apps/devflow/flow/flow.service.go [hunk 2]
```diff
@@ -577,8 +577,10 @@ func (s *Service) GetDevFlowInfo(ctx context.Context, req *pb.GetDevFlowInfoRequ
 		work.AddFunc(func(locker *limit_sync_group.Locker, i ...interface{}) error {
 			appID := i[0].(uint64)
 			appDto := appInfoMap[appID]
+			locker.Lock()
 			appTempBranchChangeBranchListMap[appID] = make(map[string][]*pb.ChangeBranch)
 			appTempBranchCommitMap[appID] = make(map[string]*apistructs.Commit)
+			locker.Unlock()
```
Comment (sfwn) [inline internal/apps/devflow/flow/flow.service.go:583]: the locked scope is too small.
`appTempBranchChangeBranchListMap` can be overwritten in L584 by another async func.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/apps/devflow/flow/flow.service.go (hunk 1 above)
Comment (littlejiancc) [reply internal/apps/devflow/flow/flow.service.go:935]: done

File: internal/apps/devflow/flow/flow.service.go (hunk 2 above)
Comment (littlejiancc) [reply internal/apps/devflow/flow/flow.service.go:583]: appTempBranchChangeBranchListMap inL584 already locked

**Review by sfwn — APPROVED** [review]

**Pull request merged**